platform/entanglement/src/services/CopyTask.js

```diff
@@ -33,36 +33,48 @@ define(
          *
          * @param domainObject The object to copy
          * @param parent The new location of the cloned object tree
-         * @param persistenceService
          * @param $q
-         * @param now
          * @constructor
          */
-        function CopyTask (domainObject, parent, persistenceService, $q, now){
+        function CopyTask (domainObject, parent, policyService, $q){
             this.domainObject = domainObject;
             this.parent = parent;
+            this.firstClone = undefined;
             this.$q = $q;
             this.deferred = undefined;
-            this.persistenceService = persistenceService;
+            this.policyService = policyService;
             this.persisted = 0;
-            this.now = now;
             this.clones = [];
         }
 
         function composeChild(child, parent) {
             //Once copied, associate each cloned
             // composee with its parent clone
-            child.model.location = parent.id;
-            parent.model.composition = parent.model.composition || [];
-            return parent.model.composition.push(child.id);
+
+            parent.getModel().composition.push(child.getId());
+
+            //Check if the object being composed is a link
+            if (!child.getCapability("location").isLink()) {
```

Comment:
It seems like this check is intended to distinguish cloned objects (whose `location` properties will have been wiped in `cloneObjectModel`.) Since the check isn't testing for exactly that, it could be brittle later - external changes to location could break this, or even maintenance of CopyTask.

I think it would be safer to make the determination about whether or not a location will be set within the block where we've consulted with policy service. Second-best place to do this would be somewhere where we have a reference to both the original and the child (e.g. `copyComposees`), where we can compare to see if this has changed. Could pass in a `setLocation` flag or a `originalChild` object if we want to keep model changes in a small number of locations.